tests/Unit/NumericalAlgorithms/FiniteDifference/Test_HighOrderFluxCorrection.cpp

```diff
@@ -252,24 +422,47 @@ void test(const fd::DerivativeOrder correction_order) {
         -1.0 / (logical_coords.get(0)[1] - logical_coords.get(0)[0]);
     evolution::dg::subcell::add_cartesian_flux_divergence(
         make_not_null(&get(get<Scalar0>(flux_divergence))), one_over_delta_xi,
-        inv_jacobian, get(get<Scalar0>(corrections_in_dim)), mesh.extents(), d);
+        get(det_inv_jacobian), get(get<Scalar0>(corrections_in_dim)),
+        subcell_mesh.extents(), d);
     for (size_t i = 0; i < Dim; ++i) {
       evolution::dg::subcell::add_cartesian_flux_divergence(
           make_not_null(&get<Vector0<Dim>>(flux_divergence).get(i)),
-          one_over_delta_xi, inv_jacobian,
-          get<Vector0<Dim>>(corrections_in_dim).get(i), mesh.extents(), d);
+          one_over_delta_xi, get(det_inv_jacobian),
+          get<Vector0<Dim>>(corrections_in_dim).get(i), subcell_mesh.extents(),
+          d);
     }
   }
 
   // With high-order corrections roundoff can accumulate.
   Approx custom_approx = Approx::custom().epsilon(5.e-12);
+  if (not aligned_coordinates and Dim == 3) {
+    // In the case of non aligned coordinates and 3-D, we are truly
+    // testing a non-aligned grid which does not have a uniform
+    // Jacobian. As such, the error is limited by the FD grid's
+    // resolution of this non-uniformity. For the lower order tests,
+    // we use very few grid points, resulting in a large error.
+    custom_approx = Approx::custom().epsilon(2.e-2);
```

Comment:
This is an extremely large error. Can't you choose a curved grid that isn't so massively underresolved? This level of error makes me worry that you are getting the wrong answer.